pkg/pool-utils/contracts/lib/PoolRegistrationLib.sol

```diff
@@ -41,6 +41,44 @@ library PoolRegistrationLib {
         // as tokens which are added to the Pool after deployment are always added to the end of the array.
         InputHelpers.ensureArrayIsSorted(tokens);
 
+        return _registerPool(vault, specialization, tokens, assetManagers);
+    }
+
+    function registerComposablePool(
+        IVault vault,
+        IVault.PoolSpecialization specialization,
+        IERC20[] memory tokens,
+        address[] memory assetManagers
+    ) internal returns (bytes32) {
+        // The Vault only requires the token list to be ordered for the Two Token Pools specialization. However,
+        // to make the developer experience consistent, we are requiring this condition for all the native pools.
+        //
+        // Note that for Pools which can register and deregister tokens after deployment, this property may not hold
+        // as tokens which are added to the Pool after deployment are always added to the end of the array.
+        InputHelpers.ensureArrayIsSorted(tokens);
+
+        IERC20[] memory composableTokens = new IERC20[](tokens.length + 1);
+        // We insert the Pool's BPT address into the first position.
```

Comment:
We need to explain _why_ we're doing this (as long as it is never removed, BPT will always be at the zero index regardless of how many tokens are added and removed because of how the Vault performs that mutation, which is useful information to have as we can remove it by dropping the first elem and add it back by doing the opposite, without having to search for it).

Reply:
We could potentially fuzz this by deploying a Vault, calling this function with some initial tokens, and then repeatedly calling add or remove with arbitrary tokens. The BPT should always be at the first index.

Constructing such a test might be a bit tricky, but I have some ideas on how to do it if you think this is worth it.

Reply:
I've added a (pretty janky) fuzz test for this. Let me know if your ideas on how to do this are more elegant.